moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1421,12 +1440,20 @@ bool TrajectoryExecutionManager::executePart(std::size_t part_index)
                          context.trajectory_parts_[longest_part].multi_dof_joint_trajectory.points.size()))
           longest_part = i;
       }
-      expected_trajectory_duration = std::max(d, expected_trajectory_duration);
-    }
-    // add 10% + 0.5s to the expected duration; this is just to allow things to finish propery
 
-    expected_trajectory_duration = expected_trajectory_duration * allowed_execution_duration_scaling_ +
-                                   ros::Duration(allowed_goal_duration_margin_);
+      // use controller-specific values if defined
+      const double current_scaling = controller_allowed_execution_duration_scaling_.count(context.controllers_[i]) > 0 ?
```

Comment:
This is inefficient, as the map is searched two times: once for count() and once for access.
Better use find() to search once and subsequently compare the retrieved iterator against end().